### PR TITLE
Revert forcing Agent version to be 8.16.0

### LINKED ISSUE #5567: Revert 8.16.0 agent package version on main
PR https://github.com/elastic/elastic-agent/pull/5565 has been created to allow green CI run until the issues with version 9.0.0 are resolved.
The purpose of this issue is to revert the hack once it's no longer necessary

### PR DESCRIPTION
**Pull request by ycombinator**
This PR reverts the changes made in #5565 as it is now possible to create `9.0.0-SNAPSHOT` deployments on ESS prod in the CFT region.

Resolves #5567

### CODE REVIEW

**Review by pierrehilbert — APPROVED** [review]

**Comment by ycombinator** [comment]
The issue seems to be that we are still creating an `8.16.0` deployment in ESS for some reason: https://buildkite.com/elastic/elastic-agent-extended-testing/builds/3296#01924d3f-01d4-4525-ae48-c1ddb2ded469/144-348

```
>>> Creating cloud stack 8.16.0-SNAPSHOT [stack_id: 8160-SNAPSHOT]
--
  | >>> Created cloud stack 8.16.0-SNAPSHOT [stack_id: 8160-SNAPSHOT, deployment_id: b463730b54b2456281dec88c341576a8]
  | >>> Waiting for cloud stack 8.16.0-SNAPSHOT to be ready [stack_id: 8160-SNAPSHOT, deployment_id: b463730b54b2456281dec88c341576a8]
```

**Comment by ycombinator** [comment]
Now a bunch of tests are failing CI with:

```
        	Error:      	Received unexpected error:
        	            	failed to find a previous minor on the version list
```

I'm working on a fix for this issue and will add it to this PR.

File: .package-version
Comment (pchila) [reply .package-version:1]: This normally is handled by the automation ---> https://github.com/elastic/elastic-agent/pull/5588/
since it would be a chicken and egg problem I guess we can update `.package.version` here and close the other PR once we merge this

**Comment by pchila** [comment]
> Now a bunch of tests are failing CI with:
> 
> ```
>         	Error:      	Received unexpected error:
>         	            	failed to find a previous minor on the version list
> ```
> 
> I'm working on a fix for this issue and will add it to this PR.

This happens because in our test where we look for a previous minor (still not convinced it is really needed) we don't handle the case of a new major with minor and patch version set to 0.

This would be a good opportuniti to get rid of the getPreviousMinor() if possible

**Review by rowlandgeoff — APPROVED** [review]
LGTM as long as CI passes 😄

**Pull request merged**
